gcc/rust/backend/rust-compile-expr.cc

```diff
@@ -375,6 +375,12 @@ CompileExpr::visit (HIR::LlvmInlineAsm &expr)
   ctx->add_statement (asm_codegen.tree_codegen_asm (expr));
 }
 
+void
+CompileExpr::visit (HIR::OffsetOf &expr)
+{
+  rust_unreachable ();
```

Comment:
I think there are maybe two ways to do this but probably you could do:

1. lookup the tyty type on the expr.get_type () HirId;

2. Then call ``tree offset_type = TyTyResolveCompile::compile (ctx, tyty_of_the_type);```

3. Then you will need to assert its a RECORD_TYPE iirc

4. You will need to iterate the record to make sure this field exists we probably should do that check on the ADTType in the type check pass upfront

5. call type_field_offset from rust-gcc.cc to get the offset and turn it into a tree using build_int_cst (size_node_type, value)

will probably do this for you